src/prototype_room_my.js

```diff
@@ -397,6 +339,66 @@ Room.prototype.executeRoom = function() {
       Game.notify(this.name + ' Under attack from ' + hostiles[0].owner.username + ' at ' + Game.time);
     }
   }
+};
+
+Room.prototype.executeRoom = function() {
+  let cpuUsed = Game.cpu.getUsed();
+  this.buildBase();
+  this.memory.attackTimer = this.memory.attackTimer || 0;
+
+  var spawns = this.find(FIND_MY_STRUCTURES, {
+    filter: function(object) {
+      return object.structureType === STRUCTURE_SPAWN;
+    }
+  });
+
+  var hostiles = this.find(FIND_HOSTILE_CREEPS, {
+    filter: this.findAttackCreeps
```

Comment:
hum yeah, looks like you just have to include it in afindAttackCreeps in proto_room_defense but it's not for this PR :) Feel free to create it or i'll do it later if i don't forget ^^'